### PR TITLE
emacsMacport: get rid of stdenv override, fix aarch64-darwin build

### PR DESCRIPTION
**Pull request by mikroskeem**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mikroskeem** [comment]
Tested with sandboxing natively, skipped sandbox on x86_64-darwin arch (via Rosetta) because it takes painfully long time to finish.

Note: to get Rosetta working under Nix with `sandbox = true`, you need to configure following:
```nix
{
  nix.sandboxPaths = [
    "/private/var/db/oah" # aot files
    "/Library/Apple"      # rosetta runtime
  ];
}
```

**Comment by mikroskeem** [comment]
OK, seems like it's unstable on aarch64.

```
Process 11508 launched: '/nix/store/9kznfvwsww0gy6f6hyhnr2pm0d5jq437-emacs-mac-27.2-8.2/Applications/Emacs.app/Contents/MacOS/Emacs' (arm64)
2021-09-18 18:21:25.430863+0300 Emacs[11508:927600] +[MTLIOAccelDevice registerDevices]: Zero Metal services found
Process 11508 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
    frame #0: 0x0000000191fef254 libobjc.A.dylib`objc_msgSend + 52
libobjc.A.dylib`objc_msgSend:
->  0x191fef254 <+52>: ldp    x17, x9, [x13], #-0x10
    0x191fef258 <+56>: cmp    x9, x1
    0x191fef25c <+60>: b.ne   0x191fef26c               ; <+76>
    0x191fef260 <+64>: eor    x10, x10, x1
Target 0: (Emacs) stopped.
(lldb) bt
* thread #1, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=1, address=0x0)
  * frame #0: 0x0000000191fef254 libobjc.A.dylib`objc_msgSend + 52
    frame #1: 0x000000019201280c libobjc.A.dylib`AutoreleasePoolPage::releaseUntil(objc_object**) + 204
    frame #2: 0x0000000191ff0e0c libobjc.A.dylib`objc_autoreleasePoolPop + 212
    frame #3: 0x000000010020d37c Emacs`mac_gui_loop + 92
    frame #4: 0x000000010020d2a8 Emacs`main + 1116
    frame #5: 0x0000000192175430 libdyld.dylib`start + 4
(lldb)
```

**Comment by mikroskeem** [comment]
Got it working, apparently launching `Emacs` inside app bundle from terminal is a bad idea.

File: pkgs/applications/editors/emacs/macport.nix [hunk 1]
```diff
@@ -27,13 +27,26 @@ stdenv.mkDerivation rec {
...
+  ];
+
   enableParallelBuilding = true;
 
-  nativeBuildInputs = [ pkg-config autoconf automake ];
+  nativeBuildInputs = [
+    pkg-config autoconf automake
+  ] ++ lib.optionals stdenv.isAarch64 [
+    sigtool
+  ];
```
Comment (Atemu) [inline pkgs/applications/editors/emacs/macport.nix:43]: Why only on aarch64?

Comment (mikroskeem) [reply pkgs/applications/editors/emacs/macport.nix:43]: It's been a while, so I'm unable to give a proper answer. But something to do with ARM macOS requiring a code signature, while Intel macOS does not.

Comment (mikroskeem) [reply pkgs/applications/editors/emacs/macport.nix:43]: IIRC emacs build system does not invoke sigtool on x86_64-darwin at all, so I didn't find it necessary to declare it as a dependency for it. Doesn't hurt to remove conditional, I just go by declaring bare minimum required to build & run the thing successfully.

**Comment by Atemu** [comment]
FYI: https://github.com/NixOS/nixpkgs/pull/155360. I'm trying to integrate your patch and got it working but both patches still crash on startup.

It's inconsistent, so perhaps, just by chance, the time you ran it via `open` worked.

**Comment by Atemu** [comment]
I wonder why our emacsMacport crashes but @railwaycat's build doesn't.

```
Fatal error 11: Segmentation fault
Backtrace:
0   Emacs                               0x0000000102389c24 deliver_thread_signal + 60
1   Emacs                               0x00000001023885d0 seed_random + 0
2   Emacs                               0x0000000102389ca0 handle_sigsegv + 64
3   libsystem_platform.dylib            0x000000018c6704e4 _sigtramp + 56
4   ???                                 0xffff80018c4e4664 0x0 + 18446603342870103652
5   libobjc.A.dylib                     0x000000018c4e4664 _objc_msgSend_uncached + 68
6   libobjc.A.dylib                     0x000000018c508f70 _ZN11objc_object17sidetable_releaseEbb + 228
7   libobjc.A.dylib                     0x000000018c4eb38c _ZN19AutoreleasePoolPage12releaseUntilEPP11objc_object + 200
8   libobjc.A.dylib                     0x000000018c4e7d68 objc_autoreleasePoolPop + 208
9   Emacs                               0x00000001024b90b4 mac_gui_loop + 92
10  Emacs                               0x00000001024b8fe0 main + 1116
11  dyld                                0x0000000102cf90f4 start + 520
Segmentation fault: 11
```

**Comment by mikroskeem** [comment]
I don't need this PR anymore, someone else feel free to pick this up.

**Pull request closed without merging**